### ISSUE TITLE
[Question] chained `onEnter` functions

### ISSUE THREAD

**Issue by johanneslumpe** [mjrussell/redux-auth-wrapper#99]
When chaining `onEnter`, always the last redirect is used. Since the `onEnter` handler does not return whether it has called the redirect function or not, it seems a bit hard to block things based on that.

One solution would be to wrap `replaceState` and keep local state there. Or am I just missing a very obvious way of dealing with this?

**Comment by mjrussell** [comment]
So the README uses the following for chaining `onEnter`:
```js
  const onEnterChain = (...listOfOnEnters) => (store, nextState, replace) => {
    listOfOnEnters.forEach(onEnter => onEnter(store, nextState, replace));
  };
```

Its not clear to me how this would only run the last `onEnter` defined. Also since JS is single threaded, if first `onEnter` took a long time to complete it shouldnt start on the second. Is maybe the problem that replace is being called multiple times and it only will redirect with the last arguments passed to replace?

**Comment by johanneslumpe** [comment]
@mjrussell That's exactly the issue. If replace gets called multiple times, the last value wins.

My solution for now looks like this:

```js
const onEnterChain = (...listOfOnEnters) => (store, nextState, replace) => {
  let redirected = false;
  const wrappedReplace = (...args) => {
    replace(...args);
    redirected = true;
  };
  listOfOnEnters.forEach((onEnter) => {
    if (!redirected) {
      onEnter(store, nextState, wrappedReplace);
    }
  });
};
```
If you agree with that approach we could add that to the docs. Another way would be that `onEnter` returns `true` or `false`, based on whether `replace` has been called internally. Then we could do:

```js
const onEnterChain = (...listOfOnEnters) => (store, nextState, replace) => {
  let redirected = false;
  listOfOnEnters.forEach((onEnter) => {
    if (!redirected) {
      redirected = onEnter(store, nextState, wrappedReplace);
    }
  });
};
```

**Comment by mjrussell** [comment]
Yup I think this is the right approach. I wouldnt want to have the onEnter returned from the auth-wrapper return a boolean because it should have the same specification as the React Router onEnter, although the "wrappedReplace" could definitely return a boolean for use in the loop. Would you mind making a PR to the docs?

**Comment by johanneslumpe** [comment]
Having only `wrappedReplace` return a boolean wouldn't work because the `onEnter` function then has to return the result of `replace`. I'll send a PR to update the docs with the first solution.

**Comment by mjrussell** [comment]
Oops yeah good catch.

**Comment by mjrussell** [comment]
Resolved in #100
